src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,175 @@
+                        this.productsMap.put(productId, productsCounter);
+                        productsCounter++;
+                    }
+                }
+                else if(content.contains(reviewKey)){
+                    String scoreValue = content.split(reviewKey)[1];
+                    reviews.add(Double.valueOf(scoreValue));
+                }
+            }
+
+            System.out.println("users: " + usersMap.size() + " products: " + productsMap.size() + " reviews: " + reviews.size());
+            this.writeCsv();
+
+        }catch (Exception e){
+            e.printStackTrace();
+        }
+    }
+
+    int getTotalReviews(){
+        return reviews.size();
+    }
+
+    int getTotalProducts(){
+        return productsMap.size();
+    }
+
+    int getTotalUsers(){
+        return usersMap.size();
+    }
+
+
+    private void writeCsv(){
+        FileWriter fileWriter = null;
+        try {
+            fileWriter = new FileWriter("movies.csv");
+
+            for(int i = 0; i < this.users.size()-1; i++){
+                int userId = this.usersMap.get(this.users.get(i));
+                fileWriter.append(String.valueOf(userId));
+                fileWriter.append(",");
+
+                int productId = this.productsMap.get(this.products.get(i));
+                fileWriter.append(String.valueOf(productId));
+                fileWriter.append(",");
+
+                fileWriter.append(String.valueOf(this.reviews.get(i)));
+
+                fileWriter.append("\n");
+            }
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        finally {
+            try {
+                assert fileWriter != null;
+                fileWriter.flush();
+                fileWriter.close();
+            }catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    List<String> getRecommendationsForUser(String userId) {
+        List<String> recommended = new ArrayList<String>();
+        List<RecommendedItem> recommendations = null;
+
+        try {
+            DataModel model = new FileDataModel(new File("movies.csv"));
+            UserSimilarity similarity = new PearsonCorrelationSimilarity(model);
+            UserNeighborhood neighborhood = new ThresholdUserNeighborhood(0.1, similarity, model);
+
+            UserBasedRecommender recommender = new GenericUserBasedRecommender(model, neighborhood, similarity);
+
+            recommendations = recommender.recommend(usersMap.get(userId), 3);
+
+        } catch (TasteException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        assert recommendations != null;
+        for (RecommendedItem recommendation : recommendations) {
+            String value = getKey((int)recommendation.getItemID());
+            recommended.add(value);
+        }
+
+        return recommended;
+
+    }
+
+    private String getKey(int value) {
```

Comment:
For this function, I understand that it returns the value associated with a key if it contains the key, right? If that's the case, you could use the function `containsKey` of your `Map`, that way, you avoid to use an explicit loop for checking the `KeySet` :)